addons/mail/models/ir_http.py

```diff
@@ -13,7 +13,11 @@ def session_info(self):
         """Override to add the current user data (partner or guest) if applicable."""
         result = super().session_info()
         store = StoreData()
-        self.env["res.users"]._init_store_data(store)
+        cids = request.httprequest.cookies.get('cids', False)
+        if cids:
+            self.env["res.users"].with_context(allowed_company_ids=[int(cid) for cid in cids.split(',')])._init_store_data(store)
+        else:
+            self.env["res.users"]._init_store_data(store)
```

Comment:
If possible, to avoid repeating code and calling `_init_store_data` twice:

```suggestion
        if cids := request.httprequest.cookies.get("cids", False):
            self = self.with_context(allowed_company_ids=[int(cid) for cid in cids.split(",")])
        self.env["res.users"]._init_store_data(store)
```

Reply:
ty for your input. No point of concern, I just had higher priority task to take care of.

Reply:
Well apparently we can't assign to self for runbot.
`[PLW0642](https://docs.astral.sh/ruff/rules/self-or-cls-assignment) coll 13: Invalid assignment to self argument in instance method`
I guess we can create a new variable for the "self with env", but that makes the diff a little longer.